hatchet_sdk/client.py

```diff
@@ -31,6 +31,42 @@ class Client:
 
 class ClientImpl(Client):
 
+    @classmethod
+    def new_from_environment(
+        cls, defaults: ClientConfig = ClientConfig(), *opts_functions
+    ):
+        config: ClientConfig = ConfigLoader(".").load_client_config(defaults)
+        for opt_function in opts_functions:
+            opt_function(config)
+
+        return cls.new_from_config(config)
+
+    @classmethod
+    def new_from_config(cls, config: ClientConfig = ClientConfig()):
```

Comment:
nit: I'd prefer simply `from_config` unless there's a good reason for the new_ prefix